client/src/features/Sign-Up/SignUp.js

```diff
@@ -0,0 +1,68 @@
+import Header from "../Header/Header";
+import Categories from "../Categories/Categories";
+import { usePopup } from "../Cart/CartPopUpContext";
+import { useState } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import "./SignUp.css"
+import CartPopUp from "../Cart/Cart";
+function SignUp({filterCategory, setFilterCategory, userId}){
+    const {openPopUp, setOpenPopUp} = usePopup();
+    const navigate = useNavigate()
+    const baseURL = process.env.REACT_APP_API_URL;
+    const [formData, setFormData] = useState({
+        first_name: '',
+        last_name: '',
+        email: '',
+        password: '',
+      });
+      const handleSubmit = async (e) => {
+        e.preventDefault();
+      
+        try {
+          const response = await axios.post(`${baseURL}/auth/register`, formData);
```

Comment:
There should be some validation that the form data is OK before sending it to the backend. Ideally you put validation in _both_ the frontend and backend, though of course that requires more work to do both.